TitleProvider/TitleProviderPool.php

```diff
@@ -17,18 +17,16 @@
 class TitleProviderPool implements TitleProviderPoolInterface
 {
     /**
-     * @var TitleProviderInterface[]
+     * @var array<string, TitleProviderInterface>
      */
-    private $providers;
+    private array $providers;
 
     /**
-     * TitleProviderPool constructor.
-     *
-     * @param TitleProviderInterface[] $providers
+     * @param iterable<string, TitleProviderInterface> $providers
      */
-    public function __construct(array $providers)
+    public function __construct(iterable $providers)
     {
-        $this->providers = $providers;
+        $this->providers = $providers instanceof \Traversable ? \iterator_to_array($providers) : $providers;
```

Comment:
This is how we handle it in the core.

```suggestion
        $this->providers = [...$providers];
```